app/services/invitations/verify_organisation_creneaux_availability.rb

```diff
@@ -0,0 +1,68 @@
+module Invitations
+  class VerifyOrganisationCreneauxAvailability < BaseService
+    def initialize(organisation_id:)
+      @organisation = Organisation.find(organisation_id)
+      Current.agent = @organisation.agents.first
+      @invitations_params_without_creneau = []
+      @grouped_invitation_params_by_category = []
+    end
+
+    def call
+      process_invitations
+      result.unavailable_params_motifs = process_invitations_params_without_creneau
+    end
+
+    private
+
+    def process_invitations
+      return unless @organisation.invitations.valid.any?
+
+      invitations_params.each do |params|
+        @invitations_params_without_creneau << params unless creneau_available?(params)
+      end
+    end
+
+    def invitations_params
+      @organisation.invitations.valid.select("DISTINCT ON (rdv_context_id) *").to_a.map do |invitation|
+        invitation.link_params.symbolize_keys
+      end
+    end
+
+    def creneau_available?(params)
+      RdvSolidaritesApi::RetrieveCreneauAvailability.call(link_params: params).creneau_availability
+    end
+
+    def process_invitations_params_without_creneau
+      @invitations_params_without_creneau.each do |invitation_params|
+        group_invitation_params_by_category(invitation_params)
+      end
+      @grouped_invitation_params_by_category
+    end
+
+    def group_invitation_params_by_category(invitation)
+      motif_category_name = MotifCategory.find_by(short_name: invitation[:motif_category_short_name]).name
+      city_code = invitation[:city_code]
+      referent_ids = invitation[:referent_ids]
+      category_params_group = find_or_initialize_category_params_group(motif_category_name)
+      category_params_group[:invations_counter] += 1
+      unless category_params_group[:city_codes].include?(city_code) || city_code.nil?
+        category_params_group[:city_codes] << city_code
+      end
+      return if category_params_group[:referent_ids].include?(referent_ids) || referent_ids.nil?
+
+      category_params_group[:referent_ids] << referent_ids
```

Comment:
on ne devrait plutôt faire 
```suggestion
      category_params_group[:referent_ids] += referent_ids
```
pour ne pas se retrouver avec un tableau de tableau je pense